MonoGame.Framework/Input/Keyboard.SDL.cs

```diff
@@ -0,0 +1,24 @@
+﻿// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Xna.Framework.Input
+{
+    public static partial class Keyboard
+    {
+        static List<Keys> _keys;
+
+        private static KeyboardState PlatformGetState()
+        {
+            var modifiers = Sdl.Keyboard.GetModState();
+            return new KeyboardState(_keys, modifiers.HasFlag(Sdl.Keyboard.Keymod.CapsLock), modifiers.HasFlag(Sdl.Keyboard.Keymod.NumLock));
```

Comment:
Enum.HasFlag() is slower and can generate garbage as it does [type checks](http://referencesource.microsoft.com/#mscorlib/system/enum.cs,877).  I would prefer to see a simpler and garbage free test such as

    (modifiers & Sdl.Keyboard.Keymod.CapsLock) == Sdl.Keyboard.Keymod.CapsLock